apps/server/src/utils/fileManagement.ts

```diff
@@ -114,3 +114,11 @@ export async function dockerSafeRename(oldPath: PathLike, newPath: PathLike) {
   await copyFile(oldPath, newPath);
   await unlink(oldPath);
 }
+
+/**
+ * finds potential file index number in our (*) format
+ * @param baseName the base name of the project file, without extention af path
+ */
+export function getProjectNumber(baseName: string): number {
+  return Number(baseName.match(/ \((\d+)\)$/)?.at(1) ?? 0);
```

Comment:
suggestion: I think regexes are a bit difficult to read and I wonder whether the corresponding code would be trivial enough that the regex is not worth it. The code might also be more robust. What do you think?